Feature/main_page

AnnaIstochnikova opened the pull request:
- Add React components for Main Page and Categories Page from previous repository.
- Fix and improve styling across all components.

- Please pay attention to the slider styles within the Categories section - still working on it.

AnnaIstochnikova committed:
Install mui/material for rendering react components by media queries

AnnaIstochnikova committed:
Add home page component and route with temporary function using static json

AnnaIstochnikova committed:
Home atoms and styles

AnnaIstochnikova committed:
Home organism and styles

AnnaIstochnikova committed:
Home template and styles

AnnaIstochnikova committed:
Cover App in browser router

AnnaIstochnikova committed:
Fix styles css

AnnaIstochnikova committed:
Update app.js

AnnaIstochnikova committed:
Add temporary files for rendering components from categories page

AnnaIstochnikova committed:
Merge branch 'feature/main_page' of https://github.com/KaczorowskaKarolina/soyummy_frontend into feature/main_page

AnnaIstochnikova committed:
Clean console.log's

AnnaIstochnikova committed:
Fix styles

AnnaIstochnikova committed:
Add functionality to button "Other categories"

AnnaIstochnikova committed:
Add routers for home page and categories page

AnnaIstochnikova committed:
Underline for chosen category

AnnaIstochnikova committed:
Separate category atom for main page with styles

AnnaIstochnikova committed:
Categories page components with styles fixes

AnnaIstochnikova committed:
Style fixes

AnnaIstochnikova committed:
Delete temporary functions and json's

krystianklimczak commented on src/components/App.jsx:
```diff
@@ -9,7 +15,25 @@ export const App = () => {
+        <Route path="/registration" element={<Registration />} />
+        <Route path="/signin" element={<SignIn />} />
+        <Route path="/" element={<SharedLayout />}> */}
+        <Route index element={<HomePage />} />
```
path `/main`

krystianklimczak commented on src/components/client/pages/categories/components/Atoms/categoryAtoms/CategoryAtom.jsx:
```diff
@@ -0,0 +1,15 @@
+import css from './CategoryAtom.module.css';
+
+const CategoryAtom = ({ title, _, isActive }) => {
+  const categoryAddress = `http://localhost:3000/categories/${title}`;
```
PORT is `3001`

krystianklimczak commented on src/components/client/pages/home/Atoms/CategoryAtom.jsx:
```diff
@@ -0,0 +1,15 @@
+import css from './CategoryAtom.module.css';
+
+const CategoryAtom = ({ title, _, isActive }) => {
+  const categoryAddress = `http://localhost:3000/categories/${title}`;
```
PORT `3001`

krystianklimczak commented on src/components/client/pages/home/Atoms/HomeAtomButton.jsx:
```diff
@@ -0,0 +1,11 @@
+
+const HomeAtomButton = ({ text }) => {
+  return (
+    <button className={css.HomeAtomButton} type="button">
+      {text}
+    </button>
```
Are we able to use our button component? ❓

AnnaIstochnikova replied:
Of course, I can change the button as soon as it's available on dev.

krystianklimczak replied:
Isn't ❓

AnnaIstochnikova replied:
It was enough to merge the 'dev' branch, and the button appeared.

krystianklimczak commented on src/components/client/pages/home/Atoms/HomeAtomLinkButton.jsx:
```diff
@@ -0,0 +1,6 @@
+const HomeAtomLinkButton = ({ title, children }) => {
+  const categoryAddress = `http://localhost:3000/categories/${title}`;
```
PORT `3001`

krystianklimczak commented on src/components/client/pages/home/Organisms/HomeOrganism.jsx:
```diff
@@ -0,0 +1,58 @@
+  ));
+};
+
+function CheckMediaQuery() {
+  const matchesMobile = useMediaQuery('(min-width:20em)');
+  const matchesTablet = useMediaQuery('(min-width:30em)');
+  const matchesDesktop = useMediaQuery('(min-width:48em)');
+
+  if (matchesDesktop) {
+    return 4;
+  }
+  if (matchesTablet) {
+    return 2;
+  }
+  if (matchesMobile) {
+    return 1;
+  }
+}
```
Declare this helper outside component 💡

krystianklimczak commented on src/components/client/pages/home/Organisms/HomeOrganism.jsx:
```diff
@@ -0,0 +1,58 @@
+  const countOfElement = CheckMediaQuery();
+
+  return recipesByCategory.map(recipeByCategory => (
+    <ul className={css.HomeOrganism} key={nanoid()}>
+      <CategoryAtom title={recipeByCategory.title} />
+      <ul className={categoryPageCss.RecipeTemplate}>
+        {recipeByCategory.recipes.map((recipe, index) => {
+          for (let i = 0; i < countOfElement; i++) {
+            if (index < countOfElement) {
+              return (
+                <RecipeOrganism
+                  className={css.HomeOrganismButton}
+                  key={nanoid()}
+                  imageSource={recipe.thumb}
+                  title={recipe.title}
+                />
+              );
+            }
+          }
+          return null;
+        })}
+      </ul>
+      <HomeAtomLinkButton title={recipeByCategory.title}>
+        <HomeAtomButton text={'See all'} />
+      </HomeAtomLinkButton>
+    </ul>
```
Ugly component 😢

AnnaIstochnikova replied:
I've changed it, let me know if it looks better now.

AnnaIstochnikova replied:
Changed again, now I hope it's pretty.

krystianklimczak reviewed (changes requested):
@AnnaIstochnikova 
Small changes described in code 💡 

@KaczorowskaKarolina waiting for redux + backend server 👍

AnnaIstochnikova committed:
Global style for scrollbar

AnnaIstochnikova committed:
Fix styles foe slider

AnnaIstochnikova committed:
Fix ports 3000=>3001

AnnaIstochnikova committed:
Helper for checking media query through React

AnnaIstochnikova committed:
Create a separate HomeMolecule to make the code more readable

AnnaIstochnikova committed:
Change route

AnnaIstochnikova committed:
Fix HomeMolecule function

AnnaIstochnikova committed:
Merge branch 'dev' of https://github.com/KaczorowskaKarolina/soyummy_frontend into feature/main_page

AnnaIstochnikova committed:
Change "See All" button using template

AnnaIstochnikova commented:
@krystianklimczak 

Changed Home Molecule so the map function is outside the component's return.
Fixed ports from 3000 to 3001.
Changed the 'Load more' button using a reusable component.

krystianklimczak commented:
@AnnaIstochnikova  ask for review again ✌️

KaczorowskaKarolina commented on src/components/App.jsx:
```diff
@@ -1,10 +1,12 @@
 
-import { StartPage } from '../pages/startPage/StartPage.jsx';
-import { SignIn } from '../pages/signIn/SignIn.jsx';
-import { Registration } from '../pages/registration/Registration.jsx';
```
you don't use Registration in routes so please comment this @JakubLawnicki 🙏

KaczorowskaKarolina commented on src/components/client/pages/categories/CategoriesPage.module.css:
```diff
@@ -0,0 +1,9 @@
+.CategoriesPage {
+  /* Delete this style if the root/body has a max-width */
+  max-width: 1600px;
```
@JakubLawnicki  please use rem not px 🙏

KaczorowskaKarolina commented on src/components/client/pages/categories/components/Atoms/categoryAtoms/CategoryAtom.module.css:
```diff
@@ -0,0 +1,28 @@
+    position: absolute;
+    width: 100%;
+    height: 0.5rem;
+    bottom: -35px;
```
@JakubLawnicki here too rem not px

KaczorowskaKarolina commented on src/components/client/pages/categories/components/Atoms/recipeAtoms/RecipeImageAtom.module.css:
```diff
@@ -0,0 +1,4 @@
+.RecipeImageAtom {
+  border-radius: 0.5rem;
```
@JakubLawnicki for border-radius it is not recommended to convert the px value to rem. 💡

KaczorowskaKarolina commented on src/components/client/pages/categories/components/Molecules/categoryMolecules/CategoryMolecule.module.css:
```diff
@@ -0,0 +1,32 @@
+        position: absolute;
+        width: 100%;
+        height: 0.2em;
+        bottom: -35px;
```
@JakubLawnicki here to change px to rem 🙏

KaczorowskaKarolina commented on src/components/client/pages/categories/components/Organisms/categoryOrganisms/CategoryOrganism.module.css:
```diff
@@ -0,0 +1,23 @@
+  hr {
+    position: relative;
+    border-color: var(--shoppingListBorder);
+    margin-top: -60px;
```
@JakubLawnicki and here to... 🙏

KaczorowskaKarolina commented on src/components/client/pages/categories/components/Organisms/categoryOrganisms/CategoryOrganism.module.css:
```diff
@@ -0,0 +1,23 @@
+    border-color: var(--shoppingListBorder);
+    margin-top: -60px;
+    width: 112%;
+    border-width: 1px;
```
@JakubLawnicki and here to... 🙏

KaczorowskaKarolina reviewed (commented):
@JakubLawnicki small changes 🙏 

I know the rem unit works almost identically to em, with the difference that the value is not inherited from the parent element. Its value is taken from the highest element in the HTML document tree. But wherever possible, please change the units from px to rem, except for border-radius, as I wrote.

I switched to your branch and nothing is displayed even though I changed the route. I installed the dependencies but still nothing.

Please check it.

AnnaIstochnikova committed:
Fix units: rem and px

AnnaIstochnikova committed:
Add transition to images on hover

AnnaIstochnikova commented:
@KaczorowskaKarolina I've fixed px to rem where it's needed, and changed border-radius to px. Also, I've added some improvements to the images appearance when hovered over.
